### ISSUE TITLE
Collect logging and metric requirements for our services layer. 

### ISSUE THREAD

**Issue by zachferland** [3box/3box#508]
Its a good time to review if we are getting what we want out of our metrics and if we have any new requirements. Metrics and logging are very similar, it would be nice to aggregate our logs and build out some simple logging infra, for saving and search logs. Some metrics are already aggregated. From these requirements we can spec the lower aggregation and tooling approach.

**Comment by zachferland** [comment]
In the POC implementation was already able to implement some of this for logging, and for metrics we have basic organization and funnels for the data at the lowest level. 

Logging - ECS makes gathering logs from all containers pretty simple, currently configured to get aggregated in cloudwatch. From there you can write additional services that stream them to other logging services for more functionality (search, visuals, alerts, etc). We can still us different logging libs in our services, but making sure they end up in stdout is simplest for these funnels.

For metrics we do have basic setup, that aggregate metrics in one place. So the base of the funnel is probably good, maybe have to changed how the data is described, but architecture would be similar, and may have change tooling/descriptiveness at higher levels. We will get better infra metrics for free as well with ecs, this was more in reference to our other metrics (segment).

Basically the outcome is that from an arch/scaling perspective we have the base in place for now and it doesn't have to affect any implementation. We should separate improving metrics from scaling epics so that we can prioritize independently, so at moment not going deeper on this yet.

@michaelsena any thoughts, and just pinging to say we can address both things separately, that any metric requirements should not effect current base scaling arch

**Comment by oed** [comment]
@zachferland If we move over to the new infra setup w/ only one node, would we use this or the existing setup for logging? I guess we could do it in stages?

**Comment by zachferland** [comment]
infra would be the same that this level, with one or many nodes, logs would be saved and collected in cloudwatch, which we can then connect other tools  as well. Logging from containers to cloudwatch is already implemented

**Comment by oed** [comment]
Right, I was more asking about how this maps to our current way of doing metrics. Would we output our metrics directly to `stdout` instead of using the analytics library we are using currently?

**Comment by zachferland** [comment]
ah yeah those metrics, yeah those would stay the same for now a this level, segment lib and then aggregated in redshift, could change but doesn't have to be decided now, so could do in stages if we want to change anything

**Comment by oed** [comment]
ok, thanks for clarifying!

**Comment by michaelsena** [comment]
Thanks @zachferland - makes sense. After we deploy one node, then maybe we can reprioritize metrics, but I agree for not the next sprint should prob be focused on completing and testing the single-node version of our scaling arch, so we can deploy to master the following sprint (17).
